Mathlib/Algebra/BigOperators/Basic.lean

```diff
@@ -569,47 +572,94 @@ theorem prod_bij {s : Finset α} {t : Finset γ} {f : α → β} {g : γ → β}
 
   The difference with `sum_bij` is that the bijection is specified with an inverse, rather than
   as a surjective injection."]
-theorem prod_bij' {s : Finset α} {t : Finset γ} {f : α → β} {g : γ → β} (i : ∀ a ∈ s, γ)
-    (hi : ∀ a ha, i a ha ∈ t) (h : ∀ a ha, f a = g (i a ha)) (j : ∀ a ∈ t, α)
+theorem prod_bij' (i : ∀ a ∈ s, κ) (j : ∀ a ∈ t, ι) (hi : ∀ a ha, i a ha ∈ t)
     (hj : ∀ a ha, j a ha ∈ s) (left_inv : ∀ a ha, j (i a ha) (hi a ha) = a)
-    (right_inv : ∀ a ha, i (j a ha) (hj a ha) = a) : ∏ x in s, f x = ∏ x in t, g x := by
-  refine' prod_bij i hi h _ _
-  · intro a1 a2 h1 h2 eq
-    rw [← left_inv a1 h1, ← left_inv a2 h2]
-    simp only [eq]
-  · intro b hb
-    use j b hb
-    use hj b hb
-    exact (right_inv b hb).symm
+    (right_inv : ∀ a ha, i (j a ha) (hj a ha) = a) (h : ∀ a ha, f a = g (i a ha)) :
+    ∏ x in s, f x = ∏ x in t, g x := by
+  refine prod_bij i hi (fun a1 h1 a2 h2 eq ↦ ?_) (fun b hb ↦ ⟨_, hj b hb, right_inv b hb⟩) h
+  rw [← left_inv a1 h1, ← left_inv a2 h2]
+  simp only [eq]
 #align finset.prod_bij' Finset.prod_bij'
 #align finset.sum_bij' Finset.sum_bij'
 
-/-- Reindexing a product over a finset along an equivalence.
-See `Equiv.prod_comp` for the version where `s` and `s'` are `univ`. -/
-@[to_additive " Reindexing a sum over a finset along an equivalence.
-See `Equiv.sum_comp` for the version where `s` and `s'` are `univ`. "]
-theorem Equiv.prod_comp_finset {ι'} [DecidableEq ι] (e : ι ≃ ι') (f : ι' → β) {s' : Finset ι'}
-    {s : Finset ι} (h : s = s'.image e.symm) : ∏ i' in s', f i' = ∏ i in s, f (e i) := by
-  rw [h]
-  refine'
-    Finset.prod_bij' (fun i' _hi' => e.symm i') (fun a ha => Finset.mem_image_of_mem _ ha)
-      (fun a _ha => by simp_rw [e.apply_symm_apply]) (fun i _hi => e i) (fun a ha => _)
-      (fun a _ha => e.apply_symm_apply a) fun a _ha => e.symm_apply_apply a
-  rcases Finset.mem_image.mp ha with ⟨i', hi', rfl⟩
-  dsimp only
-  rwa [e.apply_symm_apply]
-#align finset.equiv.prod_comp_finset Finset.Equiv.prod_comp_finset
-#align finset.equiv.sum_comp_finset Finset.Equiv.sum_comp_finset
+-- TODO: Backport arguments changes to `card_congr` and `prod_bij`
+@[to_additive]
+lemma prod_nbij (i : ι → κ) (hi : ∀ a ∈ s, i a ∈ t) (i_inj : (s : Set ι).InjOn i)
```

Comment:
What does the `n` in `nbij` stand for?

Reply:
**N**on-dependent. Bhavik came up with this name, we both don't like it. Do you have any suggestion?

Reply:
I'm inclined to go with `prod_bij'`. And just explain the difference in the docstrings using cross-refs. "See `foo_bar` for a (non)dependent variant."

Reply:
`prod_bij'` already exists! It's the version which takes in two functions that are inverse to each other on the finsets.

Reply:
Aha... arguably the name of that lemma should be closer to `equiv` than to `bij`. But maybe that's out of scope for this PR.

So let's go for `nbij` for now. But please explain the name in the docstrings.